File: src/logger.cpp
#include "../include/logger.hpp"

Logger::Logger() { message = ""; }
Logger::~Logger() {}

void Logger::log(std::string message) { std::cout << message; }
void Logger::showInput() { std::cout << input << std::endl; }
std::string Logger::getInput() const { return input; }

void Logger::getInput(std::string message) {
  std::cin >> message;
  if (message == "q") {
    exitProgram("Exiting program...");
  }

  if (message == "4" || message == "5") {
    input = message;
  } else {
    std::cout << "Invalid input. Please try again." << std::endl;
    getInput(message);
  }
}

void Logger::exitProgram(std::string message) {
  std::cout << message << std::endl;
  exit(EXIT_SUCCESS);
}

void Logger::clearScreen() {
  std::cout << "\033[2J\033[1;1H";
  std::cout.flush();
}

File: src/logger.rs
use std::io::Write;

pub struct Logger {
    input: String,
}

impl Logger {
    pub fn new() -> Self {
        Logger {
            input: String::new(),
        }
    }

    pub fn log(&self, message: &str) {
        println!("{}", message);
    }

    pub fn get_input(&self) -> &str {
        &self.input
    }

    pub fn process_input(&mut self, message: &str) {
        use std::io::{stdin, stdout};
        let mut input = String::new();

        print!("{}", message);
        stdout().flush().expect("Failed to flush stdout");
        stdin().read_line(&mut input).expect("Failed to read input");

        let input = input.trim();

        match input {
            "q" => self.exit_program("Exiting program..."),
            "4" | "5" => {
                self.input = input.to_owned();
            }
            _ => {
                println!("Invalid input. Please try again.");
                self.process_input(message);
            }
        }
    }

    pub fn exit_program(&self, message: &str) {
        println!("{}", message);
        std::process::exit(0);
    }

    pub fn clear_screen(&self) {
        print!("{esc}[2J{esc}[1;1H", esc = 27 as char);
        std::io::stdout().flush().expect("Failed to flush stdout");
    }
}
